Tribler/Test/Category/test_category.py

```diff
@@ -0,0 +1,55 @@
+import os
+from nose.tools import raises
+from Tribler.Category.Category import Category, cmp_rank
+from Tribler.Test.test_as_server import BaseTestCase
+
+
+class TriblerCategoryTest(BaseTestCase):
+
+    FILE_DIR = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+    CATEGORY_TEST_DATA_DIR = os.path.abspath(os.path.join(FILE_DIR, u"data/"))
+
+    def tearDown(self):
+        Category.delInstance()
+
+    @raises(RuntimeError)
+    def test_category_singleton(self):
+        cat = Category(custom_path=self.CATEGORY_TEST_DATA_DIR)
+        next_cat = Category(custom_path=self.CATEGORY_TEST_DATA_DIR)
+
+    def test_category_names_none_names(self):
+        cat = Category.getInstance(custom_path=self.CATEGORY_TEST_DATA_DIR)
+        cat.category_info = None
+        assert not cat.getCategoryNames()
```

Comment:
use nose.tools.assert_* instead, it will print pretty exceptions errors (like diffing a list if they are not equal and such)